vm/src/builtins.rs

```diff
@@ -312,19 +313,42 @@ fn builtin_id(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
 
 // builtin_input
 
-fn builtin_isinstance(obj: PyObjectRef, typ: PyClassRef, vm: &VirtualMachine) -> PyResult<bool> {
-    vm.isinstance(&obj, &typ)
+fn builtin_isinstance(
+    obj: PyObjectRef,
+    typ: Either<PyClassRef, PyTupleRef>,
+    vm: &VirtualMachine,
+) -> PyResult<bool> {
+    match typ {
+        Either::A(ref cls) => vm.isinstance(&obj, cls),
+        Either::B(ref tuple) => {
+            for cls_obj in tuple.elements.borrow().iter() {
+                let cls = PyClassRef::try_from_object(vm, cls_obj.clone())?;
+                if vm.isinstance(&obj, &cls)? {
+                    return Ok(true);
+                }
+            }
+            Ok(false)
+        }
+    }
 }
 
-fn builtin_issubclass(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(
-        vm,
-        args,
-        required = [(subclass, Some(vm.get_type())), (cls, Some(vm.get_type()))]
-    );
-
-    let issubclass = vm.issubclass(subclass, cls)?;
-    Ok(vm.context().new_bool(issubclass))
+fn builtin_issubclass(
+    subclass: PyClassRef,
+    typ: Either<PyClassRef, PyTupleRef>,
+    vm: &VirtualMachine,
+) -> PyResult<bool> {
+    match typ {
+        Either::A(ref cls) => vm.issubclass(&subclass, cls),
+        Either::B(ref tuple) => {
+            for cls_obj in tuple.elements.borrow().iter() {
```

Comment:
Could we use the `any` function here?

Reply:
If you are referring to the `builtin_any` function, then we would have to convert each result back into a PyObjectRef, so  `builtin_any` understands it. That seems kinda inefficiently and would 
 not really simplify the code.

If you are referring to the `any` function on a rust `Iterator`, then that doesn't handle fallible operations which we need as `isinstance`/`issubclass` can fail.

Reply:
I was referring to something like this:
```rust
tuple.elements.borrow.map(|clb_obj| obj_obj.isinstance?).any()
```
so more the rust `any` functionality. Just curiosity from my side. This code looks pretty clean.

Reply:
That won't work as `?` doesn't work within a closure like that, I wish it did work as it would make working with failable operations easier.